ProcessTable.c

```diff
@@ -55,10 +55,16 @@ static void ProcessTable_prepareEntries(Table* super) {
    Table_prepareEntries(super);
 }
 
+static void ProcessTable_aggregate(ProcessTable* this) {
+   Hashtable_foreach(this->super.table, Process_resetGroupFields, NULL);
```

Comment:
Try to include this inside the call to `ProcessTable_goThroughEntries`, as this avoids one iteration over the hash map (which is not the fastest).